kivy/uix/behaviors/compoundselection.py

```diff
@@ -492,15 +492,20 @@ def _select_range(self, multiselect, keep_anchor, node, idx):
         last_idx = self._anchor_idx
 
         if last_node is None:
-            last_idx = end
-            last_node = sister_nodes[end]
+            self.clear_selection()
```

Comment:
Looking at how windows does it, it does not clear the current selection. I.e. try ctrl selecting 1, then 3 and then shift-ctrl select 5. It will not clear the selection of 1.